lib/mapObjects/CGHeroInstance.cpp

```diff
@@ -348,16 +348,25 @@ void CGHeroInstance::initArmy(CRandomGenerator & rand, IArmyDescriptor * dst)
 	if(!dst)
 		dst = this;
 
-	int howManyStacks = 0; //how many stacks will hero receives <1 - 3>
-	int pom = rand.nextInt(99);
+	int howManyStacks = 7; //how many stacks will hero receives <1 - 7>
 	int warMachinesGiven = 0;
 
-	if(pom < 9)
-		howManyStacks = 1;
-	else if(pom < 79)
-		howManyStacks = 2;
-	else
-		howManyStacks = 3;
+	std::vector<int32_t> stacksCountChances = VLC->modh->settings.HERO_STARTING_ARMY_STACKS_COUNT_CHANCES;
+
+	const int zeroStacksAllowingValue = -1;
+	bool allowZeroStacksArmy = !stacksCountChances.empty() && stacksCountChances.back() == zeroStacksAllowingValue;
```

Comment:
Not sure if this check is needed - I expect vcmi to break on empty armies anyway.

Reply:
empty armies work in general so this feature should stay